src/debugAdapter/runtimeInterface.ts

```diff
@@ -135,18 +143,80 @@ export default class RuntimeInterface extends EventEmitter {
     // Gets the contracts compilation
     const result = await prepareContracts(workingDirectory);
 
+    // Sets the properties to use during the debugger process
+    this._mappedSources = result.mappedSources;
+    const networkId = this.getNetworkId(providerUrl);
+
     // Sets the truffle debugger options
     const options: truffleDebugger.DebuggerOptions = {
       provider: providerUrl,
       compilations: result.shimCompilations,
+      lightMode: networkId !== undefined,
     };
 
-    // Sets the properties to use during the debugger process
-    this._mappedSources = result.mappedSources;
-    this._session = await this.generateSession(txHash, options);
+    this._session = await this.generateSession(txHash, networkId, options);
+    this.serializeExternalSources();
+
     this._isDebuggerAttached = true;
   }
 
+  /**
+   * Serialize any (fetched) external sources into a temporary folder
+   * to be later opened by VS Code editor.
+   *
+   * Whenever this `Session` has been already initialized with fetch external sources,
+   * _i.e._, `fetchAndCompileForDebugger`,
+   * the corresponding sources are stored in this `Session`'s state.
+   * This method serialize these Session's sources into a temporary folder
+   * (as returned by `os.tmpdir()`).
+   *
+   * > Moreover, if the source path of a contract being serialized is a nested path,
+   * > _.e.g._, `/@openzeppelin/contracts/access/Ownable.sol`,
+   * this method creates the full folder path.
+   */
+  private serializeExternalSources() {
+    const byId = this._session!.view(this._selectors.sourcemapping.info.sources).byId;
+
+    // TODO: This guard is used so far in tests, not sure if `byId` can be undefined when running the extension.
+    if (byId === undefined) {
+      return;
+    }
+
+    for (const compilation of Object.values(byId) as {compilationId: string; source: string; sourcePath: string}[]) {
+      if (compilation.compilationId.startsWith('externalFor')) {
+        const tmp = os.tmpdir();
+        const sourcePath = tmp + '/' + compilation.sourcePath;
```

Comment:
maybe i'd use `path.join` just to keep the pattern

Reply:
Totally, fixed! thanks.